doc/object_properties.adoc

```diff
@@ -0,0 +1,252 @@
+= Object Properties
+include::../include/config.adoc[]
+:url-c-content-src: {url-engine-src}/script/common/c_content.cpp
+
+WARNING: There may be missing/incorrect information here,
+please refer to the {url-lua-api}#L7141-L7304[Lua API]
+and the {url-c-content-src}#L186-L442[relevant engine code]
+
+WARNING: This is missing the default values for everything
+
+== Properties Applying Only to Players
+
+=== `hp_max`
+* internal type: `u16`
+
+Sets the max HP of a player.
+Clamped if it goes above the `u16` max value.
+
+=== `breath_max`
+* internal type: `u16`
+
+Sets the max breath of a player.
+
+=== `eye_height`
+* internal type: `f32`
+
+How high the camera is in nodes about the player's feet.
+
+=== `zoom_fov`
+* units: `degrees`
+
+Changes the field of view of player when they zoom. Setting to `0` disables zooming.
+
+== Properties Applying to All Objects
+
+=== `physical`
+* field type: `bool`
+
+Collides with nodes that are `walkable`.
+
+=== `collide_with_objects`
+* field type: `bool`
+
+Collides with other objects.
+CAUTION: `physical` must be set to `true` for this to take effect!
+
+=== `collisionbox`
+//TIP: This could maybe be linked somewhere else.
+
+* field type: `cuboid_def`
+
+The collision box of an object. Wanted value is `{xmin, ymin, zmin, xmax, ymax, zmax}` (in nodes from the center of the object/ object position).
+
+=== `selectionbox`
+* field type: `cuboid_def`
+
+The selection box of an object, if not set, uses `collisionbox`. Wanted value is `{xmin, ymin, zmin, xmax, ymax, zmax}` (in nodes from the center of the object/ object position).
+
+=== `pointable`
+* field type: `bool`
+
+Whether the object can be pointed at.
+
+=== `visual`
+* field type: `string`
+
+Possible values:
+
+* `"cube"`: A cube that has the same size as a normal node
+* `"item"`: similar to `wielditem`, but ignores the `wield_item` property
+* `"mesh"`: Use a mesh model, see the `mesh` property 
+* `"sprite"`: A texture that is flat and always faces the player
+* `"upright_sprite"`: A texture that is flat and faces upright
+* `"wielditem"`: Used for dropped items, see the `wield_item` property
+
+What the object will look like.
+
+TIP: The `"wielditem"` value supports hardware colorization
+
+=== `mesh`
+* field type: `string`
+
+Name of the file of the mesh, only when using "mesh" visual.
+
+=== `visual_size`
+* field type: `3d_vector`
+
+Visual size multipliers. Wanted value is `{x = x, y = y, z = z}`, if `z` is not provided, `x` will be used for the `z` value.
+
+=== `textures`
+* field type: `{string, string, ...}`
+
+Wanted value is `{"texture.png", "texture.png", ...}`. The number of textures depends on `visual`:
+
+* "cube": 6 textures
+* "mesh": 1 texture for each mesh material
+* "sprite": 1 texture
+* "upright_sprite": 2 textures
+
+=== `colors`
+* field type: `ColorSpec`
```

Comment:
My intention was that field type would be the Lua value, and internal type would be the C++ value if appropriate to expose to the documentation.